core/domain/email_manager_test.py

```diff
@@ -2327,8 +2379,28 @@ def test_that_correct_rejected_voiceover_application_email_is_sent(self):
                 sent_email_model.intent,
                 feconf.EMAIL_INTENT_VOICEOVER_APPLICATION_UPDATES)
 
+    def test_can_send_emails_is_false_logs_error(self):
+        """When feconf.CAN_SEND_EMAILS is false, send_rejected_voiceover should
+        log an error.
+        """
+        observed_log_messages = []
+
+        def _mock_logging_function(msg, *args):
+            """Mocks logging.error()."""
+            observed_log_messages.append(msg % args)
+
+        with self.swap(logging, 'error', _mock_logging_function):
+            email_manager.send_rejected_voiceover_application_email(
+                self.applicant_id, 'Lesson to voiceover', 'en',
+                'A rejection message!')
+
+            expected_log_message = 'This app cannot send emails to users.'
+
+            self.assertTrue(
+                observed_log_messages[0].startswith(expected_log_message))
```

Comment:
Assert the entirety of observed_log_messages, not just the first element. In general, prefer the stronger validation check.

Also, in that spirit, why use startswith? Better to check equality of the whole thing.